### PR TITLE
UX: chat drawer increase unread channel visibility

### PR DESCRIPTION
**Pull request by dbattersby**
This change increases the visibility of unread channels to make them stand out more in drawer mode (desktop).

When a channel is unread:
- it floats to the top;
- when multiple channels are unread, they are sorted alphabetically (equal to how it’s done on mobile)
- the unread indicator blue dot moves to directly right of the channel name

### How it looks

With unread messages:

Screenshot 2024-09-04 at 2 35 40 PM

With mentions or urgent messages:

Screenshot 2024-09-04 at 2 37 06 PM

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-channel-metadata.gjs [hunk 1]
```diff
@@ -1,12 +1,7 @@
 import Component from "@glimmer/component";
 import I18n from "discourse-i18n";
-import ChatChannelUnreadIndicator from "./chat-channel-unread-indicator";
 
 export default class ChatChannelMetadata extends Component {
-  get unreadIndicator() {
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/components/chat-channel-metadata.gjs:6]: Unread indicator has moved to the channel name component

File: plugins/chat/assets/javascripts/discourse/services/chat-channels-manager.js [hunk 2]
```diff
@@ -130,11 +129,7 @@ export default class ChatChannelsManager extends Service {
         channel.isCategoryChannel && channel.currentUserMembership.following
     );
 
-    if (this.site.mobileView) {
-      return this.#sortChannelsByActivity(channels);
-    } else {
-      return channels.sort((a, b) => a?.slug?.localeCompare?.(b?.slug));
-    }
+    return this.#sortChannelsByActivity(channels);
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/services/chat-channels-manager.js:132]: Where the main change happens for channel sorting, using same activity order as mobile.

File: plugins/chat/assets/javascripts/discourse/components/channel-name/index.gjs [hunk 3]
```diff
@@ -37,43 +42,55 @@ export default class ChatChannelName extends Component {
...
+    return this.args.channel.title;
+  }
+
+  get showPluginOutlet() {
+    return this.args.channel.isDirectMessageChannel && !this.groupDirectMessage;
+  }
+
   <template>
     <div class="chat-channel-name">
-      {{#if @channel.isDirectMessageChannel}}
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/components/channel-name/index.gjs:45]: Refactored this component to make it a bit cleaner and reduce nesting.

**Review by jjaffeux — APPROVED** [review]
LGTM

**Pull request merged**